src/resources/ble_darwin.mm

```diff
@@ -38,12 +38,22 @@ explicit BLEResourceGroup(Process* process)
       : ResourceGroup(process, HostBLEEventSource::instance()) {
   }
 
+  void tear_down() override {
+    tearing_down_ = true;
+    ResourceGroup::tear_down();
+  }
+
+  bool is_tearing_down() { return tearing_down_; }
+
  protected:
   uint32_t on_event(Resource* resource, word data, uint32_t state) override {
     USE(resource);
     state |= data;
     return state;
   }
+
+ private:
+  bool tearing_down_ = false;
```

Comment:
It almost feels like we should have better support for these nested resources trees with sub-resources that are 'owned' or at least 'held on to' by another resource in the group.